src/drivers/memory.js

```diff
@@ -0,0 +1,234 @@
+// If no driver is available, we'll fall back to simple in memory storage.
+// Note that this will have considerable performance and storage
+// side-effects and data will be lost when the user reloads the page
+
+import serializer from '../utils/serializer';
+import Promise from '../utils/promise';
+import executeCallback from '../utils/executeCallback';
+import normalizeKey from '../utils/normalizeKey';
+
+var storageRepository = {};
+
+// Config the backend, using options set in the config.
+function _initStorage(options) {
+    var self = this;
+
+    var dbInfo = {};
+    if (options) {
+        for (var i in options) {
+            dbInfo[i] = options[i];
+        }
+    }
+
+    var database = storageRepository[dbInfo.name] = storageRepository[dbInfo.name] || {};
+    var table = database[dbInfo.storeName] = database[dbInfo.storeName] || {};
+    dbInfo.db = table;
+
+    self._dbInfo = dbInfo;
+    dbInfo.serializer = serializer;
+
+    return Promise.resolve();
+}
+
+// Remove all keys from the datastore, effectively destroying all data in
+// the app's key/value store!
+function clear(callback) {
+    var self = this;
+    var promise = self.ready().then(function() {
+        var db = self._dbInfo.db;
+
+        for (var key in db) {
+            if (db.hasOwnProperty(key)) {
+                delete db[key];
+            }
+        }
+    });
+
+    executeCallback(promise, callback);
+    return promise;
+}
+
+// Retrieve an item from the store. Unlike the original async_storage
```

Comment:
I don't think a lot of these comments are needed. Some of them seem copy+pasted and not even super-relevant to the in-memory driver.